### PR TITLE
docs(deploy): add stoat preview

### PR DESCRIPTION
**Pull request by tuliren**
This is the doc update I plan to propose to Vite.

Depend on https://github.com/stoat-dev/stoat-action/pull/115.

### CODE REVIEW

File: docs/guide/static-deploy.md [hunk 1]
```diff
@@ -315,3 +315,39 @@ You can deploy your Vite app as a Static Site on [Render](https://render.com/).
...
+
+## Stoat
+
+You can preview your Vite app build by GitHub workflows on pull requests with [Stoat](https://stoat.dev/).
+
+1. Install the [Stoat App](https://github.com/apps/stoat-app).
+
+2. Append the [Stoat Action]() at the of the GitHub workflow that generates the Vite preview.
+
+   ```sh{4-6}
```
Comment (jrhizor) [inline docs/guide/static-deploy.md]: should this be yaml?

**Comment by jrhizor** [comment]
Looks good. We should be able to use this as a template for other doc additions

**Review by jrhizor — APPROVED** [review]

File: docs/guide/static-deploy.md (hunk 1 above)
Comment (tuliren) [reply docs/guide/static-deploy.md]: Right. Fixed.

**Comment by tuliren** [comment]
Submitted the PR to Vite here: https://github.com/vitejs/vite/pull/11632.

**Pull request merged**